src/lib.rs

```diff
@@ -97,9 +98,60 @@ mod accept {
     }
 }
 
+mod host {
+    use url::Url;
+
+    /// The host part of a domain (without scheme, port and path).
+    ///
+    /// This is the argument to the [`connect`](crate::connect::connect) function. Strings and string slices are
+    /// converted into Hosts automatically, as is [Url](url::Url) with the `host-from-url` feature (enabled by default).
+    #[derive(Debug)]
+    pub struct Host(String);
+
+    impl Host {
+        /// The host as string. Consumes self.
+        pub fn as_string(self) -> String {
+            self.0
+        }
+    }
+
+    impl From<&str> for Host {
```

Comment:
can we add an impl for AsRef<str> that would cover things like &String as well

Reply:
I added an `impl From<&String> for Host`. Or did you mean using AsRef directly? Couldn't get that to work with From (for size not known at compile time), but also can't think of cases where it would be needed?

Reply:
```rust
    impl<T: AsRef<str>> From<T> for Host {
        fn from(host: T) -> Self {
            Host(host.as_ref().to_string())
        }
    }
```

this would work, but because `Url` implements `AsRef<str>` as well, it ends up conflicting. So leaving it as is for now.